code-studio/src/js/pd/workshop_dashboard/attendance/workshop_attendance.jsx

```diff
@@ -0,0 +1,194 @@
+    this.context.router.replace(`/workshops/${this.props.params.workshopId}/attendance/${sessionIndex}`);
+    this.setState(this.state);
+  },
+
+  handleCancelClick: function (e) {
+    this.context.router.push(`/workshops/${this.props.params.workshopId}`);
+  },
+
+  handleSaveClick: function (e) {
+    var url = `/api/v1/pd/workshops/${this.props.params.workshopId}/attendance`;
+    var data = this.prepareDataForApi();
+    this.saveRequest = $.ajax({
+      method: 'PATCH',
+      url: url ,
+      dataType: 'json',
+      contentType: 'application/json',
+      data: JSON.stringify({pd_workshop: data})
+    }).done(function () {
+      this.context.router.push('/workshops/' + this.props.params.workshopId);
+    }.bind(this));
+  },
+
+  prepareDataForApi: function () {
+    // Convert to {session_attendances: [session_id, attendances: [user_id or email]]}
+    return {
+      session_attendances: this.state.sessionAttendances.map(function (sessionAttendance) {
+        return {
+          session_id: sessionAttendance.session.id,
+          attendances: sessionAttendance.attendance.filter(function (attendance) {
+            return attendance.attended;
+          }).map(function (attendance) {
+            if (attendance.user_id) {
+              return {id: attendance.user_id};
+            }
+            // For admin-override mode, the user may not have an account so use email.
+            // In that case, an account will be created in the backend, invited by the admin.
+            return {email: attendance.email};
+          })
+        };
+      })
+    };
+  },
+
+  handleAttendanceChange: function (i, value) {
+    this.state.sessionAttendances[this.activeSessionIndex()].attendance[i].attended = value;
+    this.setState(this.state);
+  },
+
+  activeSessionIndex: function () {
+    return parseInt(this.props.params.sessionIndex, 10) || 0;
+  },
+
+  handleAdminOverrideClick: function () {
+    this.state.adminOverride = !this.state.adminOverride;
+    this.setState(this.state);
+  },
+
+  renderAdminControls: function () {
+    if (!this.state.adminActions) {
+      return null;
+    }
+    var toggle_class;
+    var style;
+    if (this.state.adminOverride) {
+      toggle_class = "fa fa-toggle-on fa-lg";
+      style = {backgroundColor: '#f5f5dc'}; // Light green
+    } else {
+      toggle_class = "fa fa-toggle-off fa-lg";
+      style = {backgroundColor: '#f5f5f5'}; // Light gray
+    }
+    return (
+      <Row>
+        <Col sm={10} style={{padding: 10}}>
+          <span style={style}>Admin: allow counting attendance for teachers not in the section? &nbsp;</span>
+          <i className={toggle_class} style={{cursor:'pointer'}} onClick={this.handleAdminOverrideClick} />
+        </Col>
+      </Row>
+    );
+  },
+
+  render: function () {
+    if (this.state.loading) {
+      return <i className="fa fa-spinner fa-pulse fa-3x" />;
+    }
+
+    let isReadOnly = this.state.workshopState == 'Ended';
```

Comment:
s/==/===

nit: Also, should we have an enum of workshopStates so that this could instead be `workshopState === WorkshopState.Ended`?